specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+      "description": "List of managed rule set definitions available for use in a policy.",
+      "properties": {
+        "value": {
+          "description": "List of managed rule set definitions.",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleSetDefinition"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to retrieve next set of managed rule set definitions."
+        }
+      }
+    },
+    "ManagedRuleSetDefinition": {
+      "description": "Describes a managed rule set definition.",
+      "properties": {
+        "properties": {
+          "description": "Describes managed rule set definition properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ManagedRuleSetDefinitionProperties"
+        },
+        "sku": {
+          "description": "The pricing tier (defines a CDN provider, feature list and rate) of the CdnWebApplicationFirewallPolicy.",
+          "$ref": "./cdn.json#/definitions/Sku"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./cdn.json#/definitions/Resource"
+        }
+      ]
+    },
+    "ManagedRuleSetDefinitionProperties": {
+      "description": "Properties for a managed rule set definition.",
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Provisioning state of the managed rule set."
+        },
+        "ruleSetType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of the managed rule set."
+        },
+        "ruleSetVersion": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Version of the managed rule set type."
+        },
+        "ruleGroups": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ManagedRuleGroupDefinition"
+          },
+          "description": "Rule groups of the managed rule set."
+        }
+      }
+    },
+    "ManagedRuleGroupDefinition": {
+      "description": "Describes a managed rule group.",
+      "properties": {
+        "ruleGroupName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Name of the managed rule group."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the managed rule group."
+        },
+        "rules": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ManagedRuleDefinition"
+          },
+          "description": "List of rules within the managed rule group."
+        }
+      }
+    },
+    "ManagedRuleDefinition": {
+      "description": "Describes a managed rule definition.",
+      "properties": {
+        "ruleId": {
```

Comment:
Same comment as previous ruleId. Should it have any constraints?

Reply:
This is all discovered via GET ManagedRuleSet API and when we have more clients like PS, CLI etc. customers will directly plumb things from the GET to the PUT. We will look into adding more constraints to the swagger in the "stable" version once we have a better understanding of the constraints. Things are in active development right now and are subject to change.